app/src/main/java/com/cornellappdev/uplift/ui/components/onboarding/PhotoPicker.kt

```diff
@@ -0,0 +1,131 @@
+package com.cornellappdev.uplift.ui.components.onboarding
+
+import android.net.Uri
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.PickVisualMediaRequest
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.ElevatedCard
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import coil.compose.rememberAsyncImagePainter
+import com.cornellappdev.uplift.util.GRAY01
+import com.cornellappdev.uplift.R
+
+/***
+ * Component that displays a camera icon for the user to select a photo for their profile
+ *  @param onPhotoSelected: function to call when the user selects a photo. Takes in uri parameter
+ */
+@Composable
+fun PhotoPicker(onPhotoSelected: (Uri) -> Unit){
+    // State to store the selected image URI
+    var selectedImageUri by rememberSaveable { mutableStateOf<Uri?>(null) }
```

Comment:
Oh one key structural note--I think it's best if we pipeline the `selectedImageUri` as a parameter. Then it'll be the job of whoever's using the component to pipeline that in.

You could imagine that this will be useful in a scenario in which the user ALREADY has a pfp that we are fetching from, like, backend, for example. We'd need some way to automatically show that as soon as the component is shown, hence the input arg.

I really love the `onPhotoSelected` though, that's the key of how we want to send data back up. The new parameter you'll make, `selectedImageUri: Uri? = null` or something like that, will be how we send data down.

Reply:
Sounds good, I'll be adding that parameter